app/services/analytics_service/celery_worker.py

```
@@ -0,0 +1,202 @@
+    def setup_signal_handlers(self):
+        """Setup signal handlers for graceful shutdown"""
+        signal.signal(signal.SIGTERM, self.signal_handler)
+        signal.signal(signal.SIGINT, self.signal_handler)
+    
+    def signal_handler(self, signum, frame):
+        """Handle shutdown signals"""
+        logger.info(f"Received signal {signum}, initiating graceful shutdown...")
+        if self.worker:
+            self.worker.stop()
+        sys.exit(0)
+    
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)